src/chocolatey/infrastructure.app/configuration/ChocolateyConfiguration.cs

```diff
@@ -67,9 +67,14 @@ public ChocolateyConfiguration()
         /// <exception cref="System.Runtime.Serialization.SerializationException">One or more objects in the class or child classes are not serializable.</exception>
         public void start_backup()
         {
+            if (_configurationBackups == null)
+            {
+                _configurationBackups = new Stack<ChocolateyConfiguration>();

```

Comment:
Any reason why you have this null check and doesn't create the stack until the first time a backup is started?

Instead of initializing the stack immediately on the creation of the class?

Reply:
Doing it that way causes issues for basically every integration test we have in the repo. Those tests use `deep_copy()` on the configuration and use that copy directly. Because `deep_copy()` uses a serialize/deserialize pattern, this property is null then even if it's set in the constructor or directly on the property.

I made an attempt to refactor them to use the backup/reset logic instead, but so far it's caused more problems than solutions. This was the simplest way to handle it.